src/plugins/discover/public/embeddable/saved_search_embeddable.test.ts

```diff
@@ -281,4 +282,33 @@ describe('saved search embeddable', () => {
     await waitOneTick();
     expect(search).toHaveBeenCalledTimes(1);
   });
+  it('should not render and fetch data more often then necessary when the embeddable title changes', async () => {
+    const search = jest.fn().mockReturnValue(
+      of({
+        rawResponse: { hits: { hits: [], total: 0 } },
+        isPartial: false,
+        isRunning: false,
+      })
+    );
+    const { embeddable } = createEmbeddable(search);
+    embeddable.render(mountpoint);
+    // wait for data fetching
+    await waitOneTick();
+    expect(search).toHaveBeenCalledTimes(1);
+
+    embeddable.reload = jest.fn();
```

Comment:
Maybe we should define it before `embeddable.render`?

Reply:
very true!